src/Microsoft.SqlTools.ServiceLayer/QueryExecution/SerializationService.cs

```diff
@@ -40,93 +41,106 @@ public override void InitializeService(IProtocolEndpoint serviceHost)
         /// <summary>
         /// Begin to process request to save a resultSet to a file in CSV format
         /// </summary>
-        internal async Task HandleSerializeStartRequest(SerializeDataStartRequestParams serializeParams,
+        internal Task HandleSerializeStartRequest(SerializeDataStartRequestParams serializeParams,
             RequestContext<SerializeDataResult> requestContext)
+        {
+            // Run in separate thread so that message thread isn't held up by a potentially time consuming file write
+            Task.Run(async () => {
+                await RunSerializeStartRequest(serializeParams, requestContext);
+            }).ContinueWithOnFaulted(async t => await SendErrorAndCleanup(serializeParams?.FilePath, requestContext, t.Exception));
+            return Task.FromResult<bool>(true);
```

Comment:
For  Task with no return value expected it suggest returning Task.CompletedTask instead of Task.FromResult<bool>(true);

Reply:
Thanks! WIll change.